plugins/inputs/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,247 @@
+			"node": summaryMetrics.Node.NodeName,
+			"name": container.Name,
+		}
+		fields := make(map[string]interface{})
+		fields["cpu_usage_nanocores"] = container.CPU.UsageNanoCores
+		fields["cpu_usage_core_nanoseconds"] = container.CPU.UsageCoreNanoSeconds
+		fields["memory_usage_bytes"] = container.Memory.UsageBytes
+		fields["memory_working_set_bytes"] = container.Memory.WorkingSetBytes
+		fields["memory_rss_bytes"] = container.Memory.RSSBytes
+		fields["memory_page_faults"] = container.Memory.PageFaults
+		fields["memory_major_page_faults"] = container.Memory.MajorPageFaults
+		fields["rootfs_available_bytes"] = container.RootFS.AvailableBytes
+		fields["rootfs_capacity_bytes"] = container.RootFS.CapacityBytes
+		fields["logsfs_avaialble_bytes"] = container.LogsFS.AvailableBytes
+		fields["logsfs_capacity_bytes"] = container.LogsFS.CapacityBytes
+		acc.AddFields("kubernetes_system_container", fields, tags)
+	}
+}
+
+func buildNodeMetrics(summaryMetrics *SummaryMetrics, acc telegraf.Accumulator) {
+	tags := map[string]string{
+		"node": summaryMetrics.Node.NodeName,
+	}
+	fields := make(map[string]interface{})
+	fields["cpu_usage_nanocores"] = summaryMetrics.Node.CPU.UsageNanoCores
+	fields["cpu_usage_core_nanoseconds"] = summaryMetrics.Node.CPU.UsageCoreNanoSeconds
+	fields["memory_available_bytes"] = summaryMetrics.Node.Memory.AvailableBytes
+	fields["memory_usage_bytes"] = summaryMetrics.Node.Memory.UsageBytes
+	fields["memory_working_set_bytes"] = summaryMetrics.Node.Memory.WorkingSetBytes
+	fields["memory_rss_bytes"] = summaryMetrics.Node.Memory.RSSBytes
+	fields["memory_page_faults"] = summaryMetrics.Node.Memory.PageFaults
+	fields["memory_major_page_faults"] = summaryMetrics.Node.Memory.MajorPageFaults
+	fields["network_rx_bytes"] = summaryMetrics.Node.Network.RXBytes
+	fields["network_rx_errors"] = summaryMetrics.Node.Network.RXErrors
+	fields["network_tx_bytes"] = summaryMetrics.Node.Network.TXBytes
+	fields["network_tx_errors"] = summaryMetrics.Node.Network.TXErrors
+	fields["fs_available_bytes"] = summaryMetrics.Node.FileSystem.AvailableBytes
+	fields["fs_capacity_bytes"] = summaryMetrics.Node.FileSystem.CapacityBytes
+	fields["fs_used_bytes"] = summaryMetrics.Node.FileSystem.UsedBytes
+	fields["runtime_image_fs_available_bytes"] = summaryMetrics.Node.Runtime.ImageFileSystem.AvailableBytes
+	fields["runtime_image_fs_capacity_bytes"] = summaryMetrics.Node.Runtime.ImageFileSystem.CapacityBytes
+	fields["runtime_image_fs_used_bytes"] = summaryMetrics.Node.Runtime.ImageFileSystem.UsedBytes
+	acc.AddFields("kubernetes_node", fields, tags)
+}
+
+func buildPodMetrics(summaryMetrics *SummaryMetrics, acc telegraf.Accumulator) {
+	for _, pod := range summaryMetrics.Pods {
+		for _, container := range pod.Containers {
+			tags := map[string]string{
+				"node":      summaryMetrics.Node.NodeName,
+				"namespace": pod.PodRef.Namespace,
+				"name":      container.Name,
+				"pod":       pod.PodRef.Name,
+				"uid":       pod.PodRef.UID,
```

Comment:
yes. everytime a pod dies and comes back it would have a new uuid. Same thing with pod name